lib/classes/PluginManager.js

```diff
@@ -0,0 +1,120 @@
+'use strict';
+
+const Utils = require('./Utils');
+const path = require('path');
+const has = require('lodash').has;
+const forEach = require('lodash').forEach;
+
+class PluginManager {
+  constructor(serverless) {
+    this._serverless = serverless;
+    this._pluginInstances = [];
+    this._commandsList = [];
+    this._commands = {};
+  }
+
+  loadAllPlugins(servicePlugins) {
+    this._loadCorePlugins();
+    this._loadServicePlugins(servicePlugins);
+  }
+
+  runCommand(command) {
+    const events = this._getEvents(command.split(' '), this._commands);
```

Comment:
I think it would be better if the command is simply an array. So the CLI has to parse the command into an array, then we don't have to do any string splitting here. Imho the plugin system can simply set that expectation